### ISSUE TITLE
Handle and report errors thrown inside of Promises

### ISSUE THREAD

**Issue by ryanoglesby-tc** [honeybadger-io/honeybadger-js#133]
It seems that errors that are triggered from inside of Promises are not caught by global `window.onerror` handlers, which means Honeybadger does not get notified of these.

I'd like to request that Honeybadger also listen for `unhandledrejection` so that these errors are also reported to Honeybadger. Does that fit into the scope of this library? I'd be willing to put together a PR too.

## What are the steps to reproduce this issue?
1. Throw an error in a Promise
2. Do not catch the error

Example:

```js
/*
  The promise below will throw `ReferenceError: nosuchfunc is not defined`

  The `onerror` listener will NOT be invoked.
  The `unhandledrejection` listener WILL be invoked.
*/
window.addEventListener('error', err => {
  console.log('window onerror', err)
})

window.addEventListener('unhandledrejection', rj => {
  console.log('unhandledrejection', rj)
})

new Promise((resolve, reject) => {
  nosuchfunc()
}).then(data => {
  // never gets called
  console.log('then:', data)
})
```

## What happens?

Errors thrown inside of promises are not caught by `window.onerror` handlers, so Honeybadger does not get notified of these.

## What were you expecting to happen?

Honeybadger is notified of any errors that bubble up from Promises.

## Any logs, error output, etc?

N/A

## Any other comments?

N/A

## What versions are you using?
**Operating System:** …
**Package Version:** …
**Browser Version:** …

**Comment by joshuap** [comment]
Thanks for documenting this, @ryanoglesby-tc! I'm open to adding this, assuming we can add a config flag to disable it, and it can call any existing event bindings after calling ours (like our `window.onerror` handler does). I think some people will want to opt-out so that they can handle promise rejections however their framework recommends.

If we make this the new default, we may need to bump the major version, since this could potentially start duplicating reports for people who have already attached to this event to report errors (or have done so in their framework).

**Comment by ryanoglesby-tc** [comment]
Ok, I'll see what I can do!

**Comment by joshuap** [comment]
Released in v2.0.0
